assembl/static2/js/app/components/debate/thread/topPostForm.jsx

```diff
@@ -0,0 +1,223 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { connect } from 'react-redux';
+import { graphql } from 'react-apollo';
+import { Row, Col, FormGroup, FormControl, Button } from 'react-bootstrap';
+import { I18n, Translate } from 'react-redux-i18n';
+
+import createPostMutation from '../../../graphql/mutations/createPost.graphql';
+import {
+  updateTopPostFormStatus,
+  updateTopPostSubject,
+  updateTopPostBody,
+  updateSubjectRemaingChars,
+  updateBodyRemaingChars
+} from '../../../actions/postsActions';
+import { displayModal, displayAlert } from '../../../utils/utilityManager';
+import { getConnectedUserId } from '../../../utils/globalFunctions';
+import { getCurrentView, getContextual } from '../../../utils/routeMap';
+
+const TEXT_INPUT_MAX_LENGTH = 140;
+const TEXT_AREA_MAX_LENGTH = 3000;
+const TEXT_AREA_ROWS = 12;
+
+const TopPostForm = ({
+  ideaId,
+  subject,
+  updateSubject,
+  body,
+  updateBody,
+  updateFormStatus,
+  isFormActive,
+  mutate,
+  refetchIdea,
+  slug,
+  subjectRemainingChars,
+  updateSubjectChars,
+  bodyRemainingChars,
+  updateBodyChars
+}) => {
+  const displayForm = (isActive) => {
+    return updateFormStatus(isActive);
+  };
+
+  const redirectToLogin = () => {
```

Comment:
Bad function name because it "redirects" (displays a modal actually) only on condition (user logged out) and else does something completely different.